### PR TITLE
feat: Select recipient from a list

### PR DESCRIPTION
**Pull request by gorandalum**
Added integration with the on-behalf-of-accounts endpoint to
retrieve a recipient list. Since we now have three different
recipient selection mode in the app (list, only phone, phone and
name) the logic in the select recipient screen ended up more
complex and needed a rewrite.

Also removed the feature-toggle now, as the select recipient
feature is now completed enough that it can be released.

### CODE REVIEW

**Comment by marius-at-atb** [comment]
Haven't looked into the code yet, but after receiving `null` from `/profile/v1/on-behalf-of/accounts`, I get this message. However the response code is 200, so this shouldn't show here?

**Comment by gorandalum** [comment]
> Haven't looked into the code yet, but after receiving `null` from `/profile/v1/on-behalf-of/accounts`, I get this message. However the response code is 200, so this shouldn't show here?

Should go away after backend makes a [fix on their end](https://mittatb.slack.com/archives/C02DL21RK7C/p1723119184460099?thread_ts=1722590697.203639&cid=C02DL21RK7C)

File: src/utils/durations.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const ONE_SECOND_MS = 1000;
+export const ONE_MINUTE_MS = 60 * ONE_SECOND_MS;
+export const ONE_HOUR_MS = 60 * ONE_MINUTE_MS;
```
Comment (marius-at-atb) [inline src/utils/durations.ts:3]: Will there be a follow-up PR to use this as source for `ONE_HOUR_MS` everywhere?
Skjermbilde 2024-08-09 kl  13 20 05

File: src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/Root_ChooseTicketRecipientScreen.tsx [hunk 2]
```diff
@@ -121,110 +45,70 @@ export const Root_ChooseTicketRecipientScreen: React.FC<Props> = ({
...
+        >
+          <TitleAndDescription themeColor={themeColor} />
+
+          <ExistingRecipientsList
+            state={state}
+            onSelect={(recipient) => {
+              animateNextChange();
+              dispatch({type: 'SELECT_RECIPIENT', recipient});
+            }}
+            onErrorOrEmpty={useCallback(() => {
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/Root_ChooseTicketRecipientScreen.tsx:67]: The other functions here are for user events (like `onPress`) which doesn't need `useCallback`. `onErrorOrEmpty` is used in a `useEffect`, and not using `useCallback` makes that effect trigger on every render.

File: src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/components/ExistingRecipientsList.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+          }}
+          style={styles.errorMessage}
+        />
+      )}
+      {recipientsQuery.status === 'success' && recipientsQuery.data?.length ? (
+        <RadioGroupSection
+          items={recipientsQuery.data}
+          itemToText={(i) => i.name}
+          itemToSubtext={(i) => i.phoneNumber}
+          keyExtractor={(i) => i.name}
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/components/ExistingRecipientsList.tsx:66]: There is a restriction that you can't create recipients with same name. But even though I agree that using account id seems better, I'll change it.

File: src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/components/PhoneAndNameInputSection.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}: {
+  state: RecipientSelectionState;
+  onChangePrefix: (p: string) => void;
+  onChangePhone: (p: string) => void;
+  onChangeName: (n: string) => void;
+  themeColor: StaticColor;
+}) => {
+  const {t} = useTranslation();
+  const styles = useStyles();
+  if (!settingPhone && !settingName) return null; // What if recipients is empty?
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/components/PhoneAndNameInputSection.tsx:31]: Yeah, that what I ended up with some other place. Just forgot to remove this comment.

File: src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/use-recipient-selection-state.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
+import {Dispatch, useReducer} from 'react';
+import {
+  ExistingRecipientType,
+  OnBehalfOfErrorCode,
+  RecipientSelectionState,
+} from '@atb/stacks-hierarchy/Root_ChooseTicketRecipientScreen/types.ts';
+
+type RecipientSelectionAction =
+  | {type: 'SET_RECIPIENTS'; recipients: ExistingRecipientType[]}
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/stacks-hierarchy/Root_ChooseTicketRecipientScreen/use-recipient-selection-state.ts:9]: Good catch! The usage of this was removed during a refactoring before committing. I'll remove it.

**Pull request merged**